src/python/pants/backend/javascript/subsystems/nodejs.py

```diff
@@ -556,12 +556,16 @@ async def prepare_corepack_tool(
                 None, ("corepack", "prepare", tool_spec if version else None, "--activate")
             ),
             description=f"Preparing configured {tool_description}.",
-            input_digest=request.input_digest,
+            # TODO: Given below, is there ever any need to supply an input_digest (e.g. package.json files) here?
+            # as far as I can see, the version is always resolved for the workspace and passed in as a CLI arg to corepack
+            # input_digest=request.input_digest,
             immutable_input_digests=environment.immutable_digest(),
-            working_directory=request.working_directory,
+            # The working directory should be the build root - if not, the output_directory will not be collected
+            # working_directory=request.working_directory,
```

Comment:
The reason for this being true should be investigated. There should not be a requirement for this tool to be ran using the builld root as cwd in practice, and pants would not want to enforce that a `package.json` file is at this position.

I suspect the `environment.corepack_home` variable is a relative path instead of an absolute one, as you noted in the comment below.

Reply:
To be clear, the reason we mount the package.json and move to its containing folder as `CWD` is to trigger corepacks default behaviour of reading the `package.json` of where it was invoked, and then have corepack persist the result in the location of `environment.corepack_home`

Reply:
Essentially `corepack_home` is missing a `{chroot}` token.

Reply:
https://github.com/pantsbuild/pants/pull/21021 Incorporpated your observation in the bugfix.